publish.js

```diff
@@ -287,7 +304,8 @@ publish.clean = function (moduleRoot, options) {
  * @param options {Object} - see defaultOptions in package.json for possible values
  */
 publish.dev = function (isTest, options) {
-    var opts = extend(true, {}, defaults, options);
+    var publishPkg = publish.getPkg(__dirname);
+    var opts = extend(true, {}, publishPkg.defaultOptions, options);
```

Comment:
We should make publish.dev more robust to ensure that it does cleanup if something goes wrong - for example, if npm publish fails, the working copy is left dirty with the modified package.json. We should make sure that we always run publish.clean even in the event of a failure.